website_product/models/product.py

```diff
@@ -0,0 +1,19 @@
+# Copyright 2018 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import api, models
+from odoo.addons.http_routing.models.ir_http import slug
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    @api.multi
+    def _compute_website_url(self):
+        super(ProductTemplate, self)._compute_website_url()
+        for product in self:
```

Comment:
Can you change this to `self.filtered(lambda p: not p.sale_ok)` and remove the statement in the loop. I think it's a bit faster with many products.